### PR TITLE
[MSPileup] Fix bad import for mspileupError

### LINKED ISSUE #11425: [MS] Define MSPileup RESTful endpoint and actions for pileup data
**Impact of the new feature**
 New WMCore microservice

**Is your feature request related to a problem? Please describe.**
Creation of a new microservice to make pileup management more efficient and easier.
Subtask of https://github.com/dmwm/WMCore/issues/9779

**Describe the solution you'd like**
In addition to the `status` and `info` standard REST APIs, we should provide a way to privileged users and systems to be able to create, read, update and delete pileup ids (CRUD operations).

My suggestion is to provide a new `pileups` endpoint, likely under `ms-pileup/data/pileups` and perform an action according to the HTTP method (and possibly data provided) executed, such as:
* GET: provided a pileup id, return its definition in the database (or empty dictionary if none).
* POST: provided an input json object, it creates a new pileup data structure. The structure needs to go under some validation; and the user need to have enough privileges for that.
* PUT: provided an input json object, it updates an already existent pileup data structure. Validation over the input data needs to be performed, and the user needs to have enough privileges for such operation as well. 
* DELETE: provided a single pileup id, it deletes that pileup structure from the database. User (or machine) needs to have super powers to execute it.

Sending and receiving data should be done with JSON objects (Content-Type: application/json).

On what concerns the validation, CRIC roles should be used for the moment. The service configuration file has to support a definition of group/roles and what actions they can perform.

**Describe alternatives you've considered**
The DELETE action should likely be done only by the system, after a grace period of inactivity has passed (after the pileup id has been active:False for some long period of time).

**Additional context**
None

### PR DESCRIPTION
**Pull request by amaltaro**
Fixes #11425 

#### Status
not-tested

#### Description
Moves the `mspileupError` function over to the Data.py module, otherwise any service other than MSPileup will fail with an import exception.
In addition, fixed other problems and recursive calls in placeholder apis.

#### Is it backward compatible (if not, which system it affects?)
YES

#### Related PRs
Bugfix for https://github.com/dmwm/WMCore/pull/11443

#### External dependencies / deployment changes
None

### CODE REVIEW

File: src/python/WMCore/MicroService/MSPileup/MSPileup.py [hunk 1]
```diff
@@ -106,7 +94,8 @@ def createPileup(self, pdict):
         :return: results of MSPileup data layer (list of dicts)
         """
         authz_match(self.role, self.group)
-        return self.mgr.createPileup(pdict)
+        # TODO: Implement logic for creating a pileup object and persist in the db
```
Comment (vkuznet) [inline src/python/WMCore/MicroService/MSPileup/MSPileup.py:97]: I'm not sure it is right decision in this PR. I suggest to not touch the logic here since it is correct delegation of creating the pileup object, i.e. the data layer should create it via its `createPileup` API. Or, at least we should discuss it and make such change in separate PR. This PR should only fix `mspileupError` loading issue and not more.

File: src/python/WMCore/MicroService/MSPileup/MSPileup.py [hunk 2]
```diff
@@ -116,7 +105,8 @@ def updatePileup(self, pdict):
         :return: results of MSPileup data layer (list of dicts)
         """
         authz_match(self.role, self.group)
-        return self.mgr.updatePileup(pdict)
+        # TODO: Implement logic for updating a pileup object and persist in the db
```
Comment (vkuznet) [inline src/python/WMCore/MicroService/MSPileup/MSPileup.py:108]: the same argument here, I'm not sure it is correct change here.

**Review by vkuznet — CHANGES_REQUESTED** [review]

File: src/python/WMCore/MicroService/MSPileup/MSPileup.py (hunk 2 above)
Comment (amaltaro) [reply src/python/WMCore/MicroService/MSPileup/MSPileup.py:108]: You are right, I didn't see it was actually a MSPileupData object (I got confused with the method names!).

File: src/python/WMCore/MicroService/MSCore/MSManager.py [hunk 3]
```diff
@@ -372,7 +372,7 @@ def post(self, doc):
             res = []
             if 'pileupName' in doc:
                 # this is create POST request
-                res = self.msPileup.create(doc)
+                res = self.msPileup.createPileup(doc)
```
Comment (amaltaro) [inline src/python/WMCore/MicroService/MSCore/MSManager.py:375]: @vkuznet it does have this fix though, so I'd rather have this PR merged.

**Comment by amaltaro** [comment]
@vkuznet there is a high chance that you only look at this again on Monday, so if unit tests come clear, I will get it merged and deployed in testbed. In any case, you already spotted an issue and I have that fixed, thanks!

**Comment by amaltaro** [comment]
test this please

**Comment by amaltaro** [comment]
Oh, the MSRuleCleaner unit test failed for two reasons:
1) testbed ms-output is down due to the problem being addressed within this PR
2) because there is the unassigned variable reference fixed in a different PR.

**Comment by amaltaro** [comment]
Valentin, given that testbed is currently broken, I wanted to get it fix before calling it a day. Anyhow, please do share any concerns that you might have here. Thanks
